tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -661,6 +663,109 @@ def _make_tf_features(self, input_feat):
       freq_inputs.append(cur_input)
     return freq_inputs
 
+class ESNCell(rnn_cell.RNNCell):
+  """Echo State Network Cell.
+
+    Based on http://www.faculty.jacobs-university.de/hjaeger/pubs/EchoStatesTechRep.pdf
+    It doesn't consider output feedback, specifically it models just the reservoir.
+
+    Here a practical guide to use Echo State Networks:
+    http://minds.jacobs-university.de/sites/default/files/uploads/papers/PracticalESN.pdf
+
+    Since at the moment TF doesn't provide a way to compute spectral radius
+    of a matrix the echo state property necessary condition `max(eig(W)) < 1` is approximated
+    scaling the norm 2 of the reservoir matrix which is an upper bound of the spectral radius.
+    See https://en.wikipedia.org/wiki/Matrix_norm#Induced_norm
```

Comment:
This probably isn't guaranteed to stay static.  Can you update the link to something like ".../Matrix_norm, the section on Induced norms."